Implement quick updates

AaronDewes opened the pull request:
This contains:

- App.yml v4 (Fixes #40)
- Quicker updates
- Fixes to set-update-channel

AaronDewes committed:
Use new backup server

AaronDewes committed:
Update backup

AaronDewes committed:
Remove debug echp

pwltr committed:
add some validation to setService

AaronDewes committed:
Merge remote-tracking branch 'origin/beta' into release/0.0.5

AaronDewes committed:
Merge remote-tracking branch 'origin/feat/new-backup-server' into release/0.0.5

AaronDewes committed:
Revert "Switch app store to beta branch"

This reverts commit 6b29a76d8119331f1730f781f44ad6c7d5f8a0e6.

AaronDewes committed:
Citadel 0.0.5

pwltr committed:
fix: update lncli binary docker container

AaronDewes committed:
Update dashboard to 0.0.13

AaronDewes committed:
Add app cli to docker-compose.yml

AaronDewes committed:
Remove app.yml v1

AaronDewes committed:
Add missing import

AaronDewes committed:
More cleanups

AaronDewes committed:
Another missing import

AaronDewes committed:
Add mount for apps

AaronDewes committed:
Remove more

AaronDewes committed:
Add dependencies.yml

AaronDewes committed:
Start making compose version dynamic

AaronDewes committed:
Fix key access

AaronDewes committed:
Fix prefix

AaronDewes committed:
Remove app-cli from docker-compose.yml

AaronDewes committed:
WIP: app.yml v4

AaronDewes committed:
Move more warnings to STDERR

AaronDewes committed:
Basic (broken) support for port maps from app.yml v4

AaronDewes committed:
Another fix

AaronDewes committed:
Better error handling

AaronDewes committed:
Better logging

AaronDewes committed:
Add missing import

AaronDewes committed:
Fix v4 port mapping code

AaronDewes committed:
Fix port map for v4

AaronDewes committed:
Another fix

AaronDewes committed:
Another fix

AaronDewes committed:
More fixes

AaronDewes committed:
Fix more

AaronDewes committed:
Debug

AaronDewes committed:
Fix path

AaronDewes committed:
Automate cleanup of containers

AaronDewes committed:
More fixes

AaronDewes committed:
More work

AaronDewes committed:
Actually handle the result properly

AaronDewes committed:
Missing import

AaronDewes committed:
Another fix

AaronDewes committed:
Merge tag 'v0.0.5-rc.2' into release/0.0.5

AaronDewes committed:
Update middleware & manager

AaronDewes committed:
Merge branch 'release/0.0.5' into feat/quick-updates

AaronDewes committed:
Fix IP assignment for v4 apps

AaronDewes committed:
More fix

AaronDewes committed:
Fix missing import

AaronDewes committed:
Missing definition

AaronDewes committed:
Update dashboard

AaronDewes committed:
Merge branch 'release/0.0.5' into feat/quick-updates

AaronDewes committed:
More fixes for app.yml v4

AaronDewes committed:
Minor fix

AaronDewes committed:
Remove unused cache files

AaronDewes committed:
Merge branch 'stable' into feat/quick-updates

AaronDewes committed:
Move app.yml v4 handling to a function

AaronDewes committed:
Multi-threading

AaronDewes committed:
Minor fix

AaronDewes committed:
Remove unused async

AaronDewes committed:
Another fix

AaronDewes committed:
Better support for dependencies.yml

AaronDewes committed:
Another fix

AaronDewes committed:
More fix

AaronDewes committed:
Open with correct permissions

AaronDewes committed:
Implement file locking

AaronDewes committed:
Improve dependency managment

AaronDewes committed:
Improve list-updates

AaronDewes committed:
Also return app names in list-updates

AaronDewes committed:
Merge branch 'beta' into feat/quick-updates

AaronDewes committed:
Remove Bitcoin and LND from dependencies.yml

AaronDewes committed:
Restart Citadel after changing update channel

AaronDewes committed:
Add quick-update trigger

AaronDewes committed:
Removal plan for app.yml v3

AaronDewes committed:
Fix removal message

AaronDewes committed:
Fix fallback source map entry

AaronDewes committed:
Add isQuickUpdate to info.json

AaronDewes committed:
Fix download URL

AaronDewes committed:
More logging

AaronDewes committed:
Simplify listing apps

AaronDewes committed:
Minor fix

AaronDewes committed:
Merge remote-tracking branch 'origin/beta' into feat/quick-updates

AaronDewes committed:
Fix redis server permissions

AaronDewes committed:
Fix set-update-channel help text

AaronDewes committed:
Citadel 0.0.6 (#59)

Co-authored-by: nolim1t - f6287b82CC84bcbd <nolim1t@users.noreply.github.com>

AaronDewes committed:
Update containers

AaronDewes committed:
Update containers in dependencies.yml

WilliamConnatser commented on app-system/sources.list:
```diff
@@ -3,9 +3,9 @@
 
 # A collection of fully FLOSS app definitions and FLOSS apps for Citadel.
-https://github.com/runcitadel/apps v3-beta
+https://github.com/runcitadel/apps v3-stable
```
Did you mean to make this change? It seems curious, lol.. just making sure!

AaronDewes replied:
No, this was a mistake, it should be `v4-dev` (and later `v4-beta` and `v4-stable`)

WilliamConnatser commented on app-system/sources.list:
```diff
@@ -3,9 +3,9 @@
 # Include them anyway, but as a separate repo.
 # Add a # to the line below to disable the repo and only use FLOSS apps.
-https://github.com/runcitadel/apps-nonfree v3-beta
+https://github.com/runcitadel/apps-nonfree v3-stable
```
Same question here

WilliamConnatser reviewed (changes requested):
Let me know about the question I had, and in the meantime I will try to test this. Not sure how I can spin up a Citadel node on the previous Citadel version so I can upgrade. I may have to enhance `citadel-dev` with this functionality? I am going to look into that momentarily.

AaronDewes committed:
Fix sources.list

pwltr commented on app/lib/manage.py:
```diff
@@ -49,26 +73,58 @@ def joinThreads(threads: List[threading.Thread]):
+def handleAppV4(app):
+    composeFile = os.path.join(appsDir, app, "docker-compose.yml")
+    os.chown(os.path.join(appsDir, app), 1000, 1000)
+    os.system("docker run --rm -v {}:/apps -u 1000:1000 {} /app-cli convert --app-name '{}' --port-map /apps/ports.json /apps/{}/app.yml /apps/{}/result.yml --services 'lnd'".format(appsDir, dependencies['app-cli'], app, app, app))
```
This pulls the image for the `app-cli` and runs containers for every v4 app I believe. It works but might lead to conflicts, or at least there's room for a speed improvement.

```
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
```

AaronDewes committed:
Pull container early

pwltr reviewed (approved):
Looks good. Only notable thing was this warning:

"WARNING: App 'ln-visualizer' has unknown dependency 'lightning'"

AaronDewes committed:
Merge remote-tracking branch 'origin/beta' into feat/quick-updates

AaronDewes committed:
Update containers

AaronDewes committed:
Fix compose isolation

AaronDewes committed:
Revert "[Feat] Docker compose isolation (#62)"

This reverts commit 1b61d525f6026be64971066a0068b887dc160552.

pwltr committed:
feat: add Citadel CLI (#33)

add commands for:
- Show status for all services
- Run a command inside a (app) container
- Switch branch / update channel
- Switch Bitcoin/Electrum/Lightning implementation
- app commands
- debug command
- Fix update / backup / starting stuck
- Edit node configs (Bitcoin Core, LND)
- Edit app configs (Nextcloud etc.)
- Memory usage / System info
- Start
- Stop
- Restart
- Reboot
- Shutdown
- List all installed apps & services
- Logs
- Version
- Help

AaronDewes committed:
Merge branch 'beta' into feat/quick-updates

AaronDewes committed:
Fix merge conflicts

Because GitHub web UI is broken

AaronDewes committed:
Fix more conflicts

AaronDewes committed:
Merge remote-tracking branch 'origin/stable' into feat/quick-updates

AaronDewes committed:
Update containers in dependencies.yml

AaronDewes committed:
Merge remote-tracking branch 'origin/beta' into feat/quick-updates

WilliamConnatser commented on app-system/sources.list:
```diff
@@ -3,7 +3,7 @@
 
 # A collection of fully FLOSS app definitions and FLOSS apps for Citadel.
-https://github.com/runcitadel/apps v3-beta
+https://github.com/runcitadel/apps v4-dev
```
Make sure to change this back

WilliamConnatser commented on app/lib/manage.py:
```diff
@@ -49,26 +73,58 @@ def joinThreads(threads: List[threading.Thread]):
-                "githubRepo": "runcitadel/core",
-                "branch": "v2"
+                "githubRepo": "runcitadel/apps",
+                "branch": "v4-stable"
```
I am doubtful that this should be hard coded

AaronDewes replied:
This is just a fallback for apps that aren't found if people try to download them, it's basically unused (Which is why this was broken for so long, it's nearly unused). It should be removed though but that's for another PR.

WilliamConnatser replied:
Cool cool, thanks for explaining. LGTM!

WilliamConnatser commented on app-system/sources.list:
```diff
@@ -3,7 +3,7 @@
 
 # A collection of fully FLOSS app definitions and FLOSS apps for Citadel.
-https://github.com/runcitadel/apps v3-beta
+https://github.com/runcitadel/apps v4-dev
```
Make sure to change this back when merging

AaronDewes replied:
Done

AaronDewes committed:
Use v4-beta branch
